tasty-plutus/src/Test/Tasty/Plutus/Context.hs

```diff
@@ -193,6 +195,21 @@ data Output
       Show
     )
 
+{-- | A valid time range for a transaction. -}
+newtype ValidRange = ValidRange POSIXTimeRange
+  deriving stock (Show)
+
+-- | Combine ValidRanges into a new range with the smallest interval that can
+--   be produced from the lower and upper bounds. This ensures that infinite
+--   ranges behave as identity values since they have no effect on the combined 
+--   range.
+instance Semigroup ValidRange where
```

Comment:
Is this actually lawful? Imagine if I fed you two disjoint ranges - the result of the operation doesn't seem to make much sense in that context.

Reply:
Admittedly, the result of two disjoint ranges would be nonsensical. However, this does satisfy the associativity and identity laws. And one could argue that attempting to create a range from two disjoint ranges is equally nonsensical as the result.

For the actual **mustValidateIn** constraint from **Ledger.Constraints**, I believe the last one added wins out (although I'd have to verify). I don't think that type of behaviour would work here.

Another option I considered was to have a version of **withValidator** that accepts the valid range as an argument. However, this seemed to create a less convenient API that further differed from **Ledger.Constraints**.

Reply:
We could solve this by basically 'glomming together' disjoint ranges to include both each other, and whatever separates them. Would that work?